packages/core/contracts/financial-templates/expiring-multiparty/Liquidatable.sol

```diff
@@ -526,8 +548,11 @@ contract Liquidatable is PricelessPositionManager {
         FixedPoint.Unsigned memory tokenRedemptionValue =
             liquidation.tokensOutstanding.mul(liquidation.settlementPrice);
 
-        // The required collateral is the value of the tokens in underlying * required collateral ratio.
-        FixedPoint.Unsigned memory requiredCollateral = tokenRedemptionValue.mul(collateralRequirement);
+        // The required collateral is the value of the tokens in underlying * required collateral ratio. The Transform
+        // Collateral requirement method applies a from the financial Product library to change the scaled the collateral
+        // requirement based on the settlement price. If no library was specified when deploying the emp then this makes no change.
+        FixedPoint.Unsigned memory requiredCollateral =
```

Comment:
What if the `sponsorCollateral < requiredCollateral` at liquidation time, but then at dispute time, the oracle price is high enough such that `requiredCollateral` decreases and `sponsorCollateral > requiredCollateral`, so the liquidator gets disputed successfully to their surprise.

This doesn't seem like a bug from the contract's point of view but is bad liquidator/disputer ux.

Reply:
Hhhmm. I'm not sure I agree. This is different from the case in the previous library with the comment that @mrice32 brought up https://github.com/UMAprotocol/protocol/pull/2185#discussion_r525366736 that identified that for the transformation of price, you need to know the timestamp of the price request. This differs because we already know the `liquidation.settlementPrice` at the liquidation timestamp, as resolved by the DVM. The price requested to the DVM and the resolution there of are all around the liquidation timestamp. 

Therefore, we are always performing the transformation considering the timestamp _of the liquidation_. What this means is that the bot that ran the liquidation/dispute (or if this was liquidation/disputed from the front end) would first need to call `transformCollateralRequirement` to get their current value of the collateral requirement when computing if a position is correctly collateralized.

Does this make sense?

Reply:
Yeah, @chrismaree, that makes sense to me.

Reply:
yeah makes sense